rllib/connectors/connector.py

```diff
@@ -404,6 +404,44 @@ def __str__(self, indentation: int = 0):
             + [c.__str__(indentation + 4) for c in self.connectors]
         )
 
+    def __getitem__(self, key: Union[str, int, type]):
+        """Returns a list of connectors that fit 'key'.
+
+        If key is a number n, we return a list with the nth element of this pipeline.
+        If key is a Connector class or a string matching the class name of a
+        Connector class, we return a list of all connectors in this pipeline matching
+        the specified class.
+
+        Args:
+            key: The key to index by
+
+        Returns: The Connector at index `key`.
+        """
+        # In case key is a class
+        if not isinstance(key, str):
+            if isinstance(key, slice):
+                raise NotImplementedError(
+                    "Slicing of ConnectorPipeline is not " "supported."
+                )
+            elif isinstance(key, int):
+                return [self.connectors[key]]
+            elif isinstance(key, type):
+                key = key.__name__
+            else:
+                raise NotImplementedError(
+                    "Indexing by {} not supported.".format(type(key))
+                )
+
+        results = []
+        for c in self.connectors:
+            if c.__class__.__name__ == key:
+                results.append(c)
+
+        if len(results) == 0:
+            raise IndexError
```

Comment:
can we simply return empty [] instead of raising error?
it feels like a common use case for someone to want to check whether a specific connector is configured for current pipeline, if not, create it.

Reply:
Got it!